File: LeetCode/cpp/1042.cpp
class Solution {
public:
    vector<int> gardenNoAdj(int N, vector<vector<int>> &paths) {
        vector<int> res(N);
        vector<vector<int>> adjacent(N);
        for (auto path:paths) {
            adjacent[path[0] - 1].push_back(path[1] - 1);
            adjacent[path[1] - 1].push_back(path[0] - 1);
        }
        for (int i = 0; i < N; ++i) {
            vector<bool> used(5);
            auto &ad = adjacent[i];
            for (auto &a:ad)
                used[res[a]] = true;
            for (int j = 1; j <= 4; ++j) {
                if (!used[j]) {
                    res[i] = j;
                    break;
                }
            }
        }
        return res;
    }
};

File: LeetCode/python3/1042.py
class Solution:
    def gardenNoAdj(self, N: int, paths: List[List[int]]) -> List[int]:
        adjacent, res = [[] for _ in range(N)], [0 for _ in range(N)]
        for path in paths:
            adjacent[path[0] - 1].append(path[1] - 1)
            adjacent[path[1] - 1].append(path[0] - 1)
        for i in range(N):
            used, ad = [False for _ in range(5)], adjacent[i]
            for a in ad:
                used[res[a]] = True
            for j in range(1, 5):
                if not used[j]:
                    res[i] = j
                    break
        return res
